File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/ProofType.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

public enum ProofType {
    INTEGRITY_PROOF_TYPE,
    SPARSE_MT_PROOF_TYPE;

    public IdentityEntitiesV2.ProofType toProto() {
        switch (this) {
            case INTEGRITY_PROOF_TYPE:
                return IdentityEntitiesV2.ProofType.IntegrityProofType;
            case SPARSE_MT_PROOF_TYPE:
                return IdentityEntitiesV2.ProofType.SparseMtProofType;
            default:
                return IdentityEntitiesV2.ProofType.UNRECOGNIZED;
        }
    }
}

File: languages/js/src/entity/identity_v2/proof_type.ts
import { ProofType as ProofTypeProto } from "../../bridge/proto/identity_entities_v2";

export enum ProofType {
  INTEGRITY_PROOF_TYPE = 0,
  SPARSE_MT_PROOF_TYPE = 1
}

export namespace ProofType {
  export function toProto(this: ProofType): ProofTypeProto {
    switch (this) {
      case ProofType.INTEGRITY_PROOF_TYPE:
        return ProofTypeProto.IntegrityProofType;
      case ProofType.SPARSE_MT_PROOF_TYPE:
        return ProofTypeProto.SparseMtProofType;
      default:
        return ProofTypeProto.UNRECOGNIZED;
    }
  }
}
